### DISCUSSION TITLE
Wire Ethernet support for the ST32F4xx devices

### DISCUSSION THREAD

**Discussion by neuronthc** [terjeio/grblHAL#213] [Ideas]
Hello Terjeio!
Great works. It's a very good idea to separate grbl from the hardware platform.
Do you planned to add support for the LwIP stack on the ST32F4xx devices?
I am try to make same on the my processor module with ST32F407VE + LAN8720A.
Thank you!!!

**Comment by terjeio** [comment]
It depends on what you mean by adding lwIP support. What I have done this far is to add protcol support on top of lwIP and use third-party drivers for the required low-level code. I am not to keen to start writing and debugging low-level code...

If there is low-level code available that can be used then adding protocol support on top of lwIP is fairly trivial as code from another driver can be used as a starting point. The [iMXRT1062](https://github.com/grblHAL/iMXRT1062/blob/master/grblHAL_Teensy4/src/enet.c) implementation would be the best as this does not use a RTOS? Some changes has to be made to driver.c as well, most can be done by copy/paste.

If you have/can make lwIP running on the board combo outside of grblHAL then that would be a good start.

BTW support for the STM32F4xx on-chip ethernet peripheral (for those that have it) is likely to be added at some point. I need hardware for debugging first, or somebody else has to add it.

Edit: fixed link to iMXRT1062 code

**Reply by neuronthc** [reply]
Thank you for fast reply!
I will take your advice and start with modification the enet.c from the IMXRT1062.
Usually I am use MDK Keil for develop and debug. I have a working board with support UDP protocol. Of course better try to add GRBL to this project, but now not clear for me how all compiled and working together.
Try to start to learn.
